susemanager/src/mgr-create-bootstrap-repo

```diff
@@ -301,104 +433,158 @@ def create_repo(label, flush, additional=[]):
 
-except ImportError as e:
-    sys.stderr.write("Unable to load module '%s'\n" % modulename)
-    sys.stderr.write(str(e) + "\n")
-    sys.exit(1)
+    return errors
 
-dryrun = options.dryrun
 
-if not options.list and not options.create:
-    options.interactive = True
 
-if options.interactive:
-    label_map = list_labels()
+#################################################################################
+### main
+#################################################################################
 
-    elabel = None
-    while True:
-        try:
-            elabel = label_map.get(int(input("Enter a number of a product label: ")), '')
-            break
-        except Exception:
-            print("Please enter a number.")
-
-    if elabel not in mgr_bootstrap_data.DATA:
-        print("'%s' not found" % elabel)
+def main():
+    # quick check to see if you are a super-user.
+    if os.getuid() != 0:
+        sys.stderr.write('ERROR: must be root to execute.\n')
         sys.exit(1)
 
-    create_repo(elabel, options.flush, additional=args)
-elif options.list:
-    list_labels()
-elif options.create:
-    if options.create not in mgr_bootstrap_data.DATA:
-        print("'%s' not found" % options.create)
+    global LOCK
+    try:
+        LOCK = rhnLockfile.Lockfile('/var/run/mgr-create-bootstrap-repo.pid')
+    except rhnLockfile.LockfileLockedException:
+        sys.stderr.write("ERROR: attempting to run more than one instance of "
+                         "mgr-create-bootstrap-repo Exiting.\n")
         sys.exit(1)
-    create_repo(options.create, options.flush, additional=args)
+
+    global BETA
+
+    opts, args, mgr_bootstrap_data = cli()
+    r = 0
+
+    if opts.auto:
+        r = generate_all(opts, mgr_bootstrap_data, additional=args)
+    elif opts.interactive:
+        label_map = list_labels(mgr_bootstrap_data)
+        if not label_map:
+            log("No products available")
+            sys.exit(0)
+
+        elabel = None
+        while True:
+            try:
+                elabel = label_map.get(int(input("Enter a number of a product label: ")), '')
+                break
+            except Exception:
+                print("Please enter a number.")
+
+        if elabel not in mgr_bootstrap_data.DATA:
+            log_error("'%s' not found" % elabel)
+            sys.exit(1)
+
+        r = create_repo(elabel, opts, mgr_bootstrap_data, additional=args)
+    elif opts.list:
+        list_labels(mgr_bootstrap_data)
+    elif opts.create:
+        if opts.create not in mgr_bootstrap_data.DATA:
+            log_error("'%s' not found" % opts.create)
+            sys.exit(1)
+        r = create_repo(opts.create, opts, mgr_bootstrap_data, additional=args)
+    releaseLOCK()
+    return r
+
+if __name__ == '__main__':
+    try:
+        sys.exit(abs(main() or 0))
+    except KeyboardInterrupt:
+        sys.stderr.write("\nProcess has been interrupted.\n")
```

Comment:
Shouldn't we release the lock as well?

Reply:
I just copied what spacewalk-repo-sync is doing and it does not release the lock in this case.